framework/sessions/__init__.py

```diff
@@ -119,13 +124,13 @@ def set_session(session):
 def create_session(response, data=None):
     current_session = get_session()
     # TODO: check if session data changed and decide whether to save the session object
-    current_session['auth_user_username'] = data['auth_user_username']
-    current_session['auth_user_id'] = data['auth_user_id']
-    current_session['auth_user_fullname'] = data['auth_user_fullname']
-    current_session['auth_user_external_first_login'] = data['auth_user_external_first_login']
-    current_session['auth_user_external_id_provider'] = data['auth_user_external_id_provider']
-    current_session['auth_user_external_id'] = data['auth_user_external_id']
-    current_session['service_url'] = data['service_url']
+    current_session['auth_user_username'] = data.get('auth_user_username')
+    current_session['auth_user_id'] = data.get('auth_user_id')
+    current_session['auth_user_fullname'] = data.get('auth_user_fullname')
+    current_session['auth_user_external_first_login'] = data.get('auth_user_external_first_login')
+    current_session['auth_user_external_id_provider'] = data.get('auth_user_external_id_provider')
+    current_session['auth_user_external_id'] = data.get('auth_user_external_id')
+    current_session['service_url'] = data.get('service_url')
```

Comment:
For myself: session are used in more places than we thought, there are more keys and there could be new ones in the future.

```
for key, value in data.items() if data else []:
    user_session[key] = value
```